kalite/inline/features/steps/inline-help.py

```diff
@@ -29,15 +33,18 @@ def step_impl(context):
 
 @then("I see a modal with step number {expected_num:d}")
 def step_impl(context, expected_num):
-    actual_num = get_modal_step_number(context)
-    assert actual_num == expected_num, "Unexpected step number '%s' found. Expected: %s" % (actual_num, expected_num)
+
+    def condition(driver):
+        number_el = driver.find_element_by_class_name(STEP_NUMBER_CLASS)
+        actual_num_text = number_el.text
+        return int(actual_num_text) == expected_num if actual_num_text else False
+
+    WebDriverWait(context.browser, 30, ignored_exceptions=NoSuchElementException).until(condition)
 
 
 @then("an element is highlighted")
 def step_impl(context):
     elem = find_css_class_with_wait(context, HIGHLIGHTED_CLASS)
-    assert elem is not None, "Couldn't find a highlighted element on the page."
```

Comment:
`find_css_class_with_wait` will raise a TimeoutError already. The assertion is redundant.